From 1b0fbda89d4a8f8223071addaab776acb62f686f Mon Sep 17 00:00:00 2001
From: helly25 <marcus.borger@snyk.io>
Date: Wed, 30 Apr 2025 03:18:02 -0700
Subject: [PATCH 1/2] Update netty.

1) io.netty:netty-common
Fixed in io.netty:netty-common @4.1.118, @4.2.0.RC3
https://cwe.mitre.org/data/definitions/1284.html
https://www.cve.org/CVERecord?id=CVE-2025-25193
https://www.first.org/cvss/calculator/4.0#CVSS:4.0/AV:L/AC:L/AT:N/PR:L/UI:N/VC:N/VI:N/VA:H/SC:N/SI:N/SA:N/E:P
https://app.snyk.io/vuln/SNYK-JAVA-IONETTY-8707740

2) io.netty:netty-handler
Fixed in io.netty:netty-handler @4.1.118.Final, @4.2.0.RC3
https://cwe.mitre.org/data/definitions/1284.html
https://www.cve.org/CVERecord?id=CVE-2025-24970
https://www.first.org/cvss/calculator/4.0#CVSS:4.0/AV:N/AC:L/AT:N/PR:N/UI:N/VC:N/VI:N/VA:H/SC:N/SI:N/SA:N
https://app.snyk.io/vuln/SNYK-JAVA-IONETTY-8707739

3) io.netty:netty-codec-http
Fixed in io.netty:netty-codec-http @4.1.108.Final
https://cwe.mitre.org/data/definitions/770.html
https://www.cve.org/CVERecord?id=CVE-2024-29025
https://www.first.org/cvss/calculator/3.1#CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:L/E:P
https://app.snyk.io/vuln/SNYK-JAVA-IONETTY-6483812

Closes #25922.

PiperOrigin-RevId: 753093711
Change-Id: I968e3a2ace748bd6ddb0a01d7f018e86100719b4
(cherry picked from commit 9d2c55613a075c10c07dc9e1e0c4e2487614c42a)
---
 MODULE.bazel       |  42 +++++++++----------
 maven_install.json | 100 ++++++++++++++++++++++++++-------------------
 2 files changed, 79 insertions(+), 63 deletions(-)

diff --git a/MODULE.bazel b/MODULE.bazel
index 11a6075175..ce8146bc2d 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -139,33 +139,33 @@ maven.install(
         "io.grpc:grpc-protobuf:1.66.0",
         "io.grpc:grpc-protobuf-lite:1.66.0",
         "io.grpc:grpc-stub:1.66.0",
-        "io.netty:netty-buffer:4.1.100.Final",
-        "io.netty:netty-codec:4.1.100.Final",
-        "io.netty:netty-codec-http:4.1.100.Final",
-        "io.netty:netty-codec-http2:4.1.100.Final",
-        "io.netty:netty-common:4.1.100.Final",
-        "io.netty:netty-handler:4.1.100.Final",
-        "io.netty:netty-handler-proxy:4.1.100.Final",
-        "io.netty:netty-resolver:4.1.100.Final",
-        "io.netty:netty-resolver-dns:4.1.100.Final",
+        "io.netty:netty-buffer:4.1.119.Final",
+        "io.netty:netty-codec:4.1.119.Final",
+        "io.netty:netty-codec-http:4.1.119.Final",
+        "io.netty:netty-codec-http2:4.1.119.Final",
+        "io.netty:netty-common:4.1.119.Final",
+        "io.netty:netty-handler:4.1.119.Final",
+        "io.netty:netty-handler-proxy:4.1.119.Final",
+        "io.netty:netty-resolver:4.1.119.Final",
+        "io.netty:netty-resolver-dns:4.1.119.Final",
         "io.netty:netty-tcnative-boringssl-static:jar:linux-aarch_64:2.0.56.Final",
         "io.netty:netty-tcnative-boringssl-static:jar:linux-x86_64:2.0.56.Final",
         "io.netty:netty-tcnative-boringssl-static:jar:osx-aarch_64:2.0.56.Final",
         "io.netty:netty-tcnative-boringssl-static:jar:osx-x86_64:2.0.56.Final",
         "io.netty:netty-tcnative-boringssl-static:jar:windows-x86_64:2.0.56.Final",
         "io.netty:netty-tcnative-classes:2.0.56.Final",
-        "io.netty:netty-transport:4.1.100.Final",
-        "io.netty:netty-transport-classes-epoll:4.1.100.Final",
-        "io.netty:netty-transport-classes-kqueue:4.1.100.Final",
-        "io.netty:netty-transport-native-epoll:jar:linux-aarch_64:4.1.100.Final",
-        "io.netty:netty-transport-native-epoll:jar:linux-x86_64:4.1.100.Final",
-        "io.netty:netty-transport-native-kqueue:jar:osx-aarch_64:4.1.100.Final",
-        "io.netty:netty-transport-native-kqueue:jar:osx-x86_64:4.1.100.Final",
-        "io.netty:netty-transport-native-unix-common:4.1.100.Final",
-        "io.netty:netty-transport-native-unix-common:jar:linux-aarch_64:4.1.100.Final",
-        "io.netty:netty-transport-native-unix-common:jar:linux-x86_64:4.1.100.Final",
-        "io.netty:netty-transport-native-unix-common:jar:osx-aarch_64:4.1.100.Final",
-        "io.netty:netty-transport-native-unix-common:jar:osx-x86_64:4.1.100.Final",
+        "io.netty:netty-transport:4.1.119.Final",
+        "io.netty:netty-transport-classes-epoll:4.1.119.Final",
+        "io.netty:netty-transport-classes-kqueue:4.1.119.Final",
+        "io.netty:netty-transport-native-epoll:jar:linux-aarch_64:4.1.119.Final",
+        "io.netty:netty-transport-native-epoll:jar:linux-x86_64:4.1.119.Final",
+        "io.netty:netty-transport-native-kqueue:jar:osx-aarch_64:4.1.119.Final",
+        "io.netty:netty-transport-native-kqueue:jar:osx-x86_64:4.1.119.Final",
+        "io.netty:netty-transport-native-unix-common:4.1.119.Final",
+        "io.netty:netty-transport-native-unix-common:jar:linux-aarch_64:4.1.119.Final",
+        "io.netty:netty-transport-native-unix-common:jar:linux-x86_64:4.1.119.Final",
+        "io.netty:netty-transport-native-unix-common:jar:osx-aarch_64:4.1.119.Final",
+        "io.netty:netty-transport-native-unix-common:jar:osx-x86_64:4.1.119.Final",
         "io.reactivex.rxjava3:rxjava:3.1.2",
         "it.unimi.dsi:fastutil:7.2.1",
         "javax.activation:javax.activation-api:1.2.0",
diff --git a/maven_install.json b/maven_install.json
index a34d36a813..a86e65563e 100644
--- a/maven_install.json
+++ b/maven_install.json
@@ -1,7 +1,7 @@
 {
   "__AUTOGENERATED_FILE_DO_NOT_MODIFY_THIS_FILE_MANUALLY": "THERE_IS_NO_DATA_ONLY_ZUUL",
-  "__INPUT_ARTIFACTS_HASH": 1867344854,
-  "__RESOLVED_ARTIFACTS_HASH": 534176767,
+  "__INPUT_ARTIFACTS_HASH": -170859570,
+  "__RESOLVED_ARTIFACTS_HASH": 1513835236,
   "conflict_resolution": {
     "com.google.api.grpc:proto-google-common-protos:2.29.0": "com.google.api.grpc:proto-google-common-protos:2.41.0",
     "com.google.auth:google-auth-library-credentials:1.6.0": "com.google.auth:google-auth-library-credentials:1.23.0",
@@ -22,8 +22,20 @@
     "com.google.j2objc:j2objc-annotations:2.8": "com.google.j2objc:j2objc-annotations:3.0.0",
     "com.google.truth:truth:1.1.2": "com.google.truth:truth:1.4.2",
     "com.google.truth:truth:1.4.0": "com.google.truth:truth:1.4.2",
+    "io.netty:netty-buffer:4.1.100.Final": "io.netty:netty-buffer:4.1.119.Final",
+    "io.netty:netty-codec-http2:4.1.100.Final": "io.netty:netty-codec-http2:4.1.119.Final",
+    "io.netty:netty-codec-http:4.1.100.Final": "io.netty:netty-codec-http:4.1.119.Final",
+    "io.netty:netty-codec-socks:4.1.100.Final": "io.netty:netty-codec-socks:4.1.119.Final",
+    "io.netty:netty-codec:4.1.100.Final": "io.netty:netty-codec:4.1.119.Final",
+    "io.netty:netty-common:4.1.100.Final": "io.netty:netty-common:4.1.119.Final",
+    "io.netty:netty-handler-proxy:4.1.100.Final": "io.netty:netty-handler-proxy:4.1.119.Final",
+    "io.netty:netty-handler:4.1.100.Final": "io.netty:netty-handler:4.1.119.Final",
+    "io.netty:netty-resolver:4.1.100.Final": "io.netty:netty-resolver:4.1.119.Final",
     "io.netty:netty-tcnative-boringssl-static:2.0.56.Final": "io.netty:netty-tcnative-boringssl-static:2.0.61.Final",
     "io.netty:netty-tcnative-classes:2.0.56.Final": "io.netty:netty-tcnative-classes:2.0.61.Final",
+    "io.netty:netty-transport-native-epoll:4.1.100.Final": "io.netty:netty-transport-native-epoll:4.1.119.Final",
+    "io.netty:netty-transport-native-unix-common:4.1.100.Final": "io.netty:netty-transport-native-unix-common:4.1.119.Final",
+    "io.netty:netty-transport:4.1.100.Final": "io.netty:netty-transport:4.1.119.Final",
     "io.opencensus:opencensus-api:0.31.0": "io.opencensus:opencensus-api:0.31.1",
     "org.checkerframework:checker-qual:3.19.0": "org.checkerframework:checker-qual:3.42.0",
     "org.mockito:mockito-core:4.3.1": "org.mockito:mockito-core:5.4.0",
@@ -538,69 +550,69 @@
     },
     "io.netty:netty-buffer": {
       "shasums": {
-        "jar": "462874b44ee782fbefec64078cda6eb8e7bf9f0e0af71a928ef4c1f2d564f7ee"
+        "jar": "32226218d854c2e51d4935a7ebe4663b72ecb9af4da881b86bf518e61aed48d3"
       },
-      "version": "4.1.100.Final"
+      "version": "4.1.119.Final"
     },
     "io.netty:netty-codec": {
       "shasums": {
-        "jar": "180a01ed67af399602e24ff1c32864e7f57f57c4a0fa5e9ab3fe9b0e5e9cf051"
+        "jar": "c497934a057444b126a86e06f2b509188e48ca6c5a5e14c182a02f1e8c8d5fdb"
       },
-      "version": "4.1.100.Final"
+      "version": "4.1.119.Final"
     },
     "io.netty:netty-codec-dns": {
       "shasums": {
-        "jar": "f7f55bee95d2467e8bda6dda980a2830b351b0dd6f0f87fe5114645035ee55ca"
+        "jar": "8893bb399849629eecaa1fa12d48aad73161721b7ce75fdeb72e6e40c549ad55"
       },
-      "version": "4.1.100.Final"
+      "version": "4.1.119.Final"
     },
     "io.netty:netty-codec-http": {
       "shasums": {
-        "jar": "326811d249cb0e5555e78e026e877834e792261c38f0666d80464426695d9590"
+        "jar": "ff48db63e44e273e677832e52cc2a1ce2670da796f22af129a8bd4c3786a2d34"
       },
-      "version": "4.1.100.Final"
+      "version": "4.1.119.Final"
     },
     "io.netty:netty-codec-http2": {
       "shasums": {
-        "jar": "23b4a74350f4cf8d41b93fb93d52b5050667d8d53fffc385672c86eab83b8749"
+        "jar": "5ea89af402b6f59715a84780c4cd0388f4e602e339a5b7b888c17ce9f5537d1a"
       },
-      "version": "4.1.100.Final"
+      "version": "4.1.119.Final"
     },
     "io.netty:netty-codec-socks": {
       "shasums": {
-        "jar": "608a453b90f8384ba4efcdc6db7f899a1f10b9ea1890954696e6cfac45ff1ba9"
+        "jar": "a85103bb1a32cf56b8d77e57f6647a859751cab6be189085df76eed18bb54c9a"
       },
-      "version": "4.1.100.Final"
+      "version": "4.1.119.Final"
     },
     "io.netty:netty-common": {
       "shasums": {
-        "jar": "d2908301f1ac6f2910900742473c15d701765d3d4467acdb1eebb9df3aa82885"
+        "jar": "991a69b1594dcad9926e268e7e94530cfdaa89feef42af26922a302d3e2e9212"
       },
-      "version": "4.1.100.Final"
+      "version": "4.1.119.Final"
     },
     "io.netty:netty-handler": {
       "shasums": {
-        "jar": "0e10e584c2e7fdf7f4804e14760ed987003f1b62ab982f62eaf13a9892793d3a"
+        "jar": "9b49be98e36bb79a2025ffb5159aec87da2720cfa545f56de39c3bc108587b1b"
       },
-      "version": "4.1.100.Final"
+      "version": "4.1.119.Final"
     },
     "io.netty:netty-handler-proxy": {
       "shasums": {
-        "jar": "686dbc2e61407f216d6cb267dd7954896f851dd34b58be3e757c5a89f20a5e67"
+        "jar": "7a35e62259a43e5dbdc2e402934b1f64084ce73aecf7b42ce647c918a6e5f617"
       },
-      "version": "4.1.100.Final"
+      "version": "4.1.119.Final"
     },
     "io.netty:netty-resolver": {
       "shasums": {
-        "jar": "c42c481c776e9d367a45cc3a67a06f65897d280334eb30b2362b8c55b7523f4f"
+        "jar": "3b6c9f9f6f360747213d154971dec1fc8852820dc1a98d055ff09d09676f8836"
       },
-      "version": "4.1.100.Final"
+      "version": "4.1.119.Final"
     },
     "io.netty:netty-resolver-dns": {
       "shasums": {
-        "jar": "4bafbd3e571ed2d9fb3f9550f8348ba37c183eda4a253fba5354166e409624c4"
+        "jar": "119e1dae4b207197ec6a4b434894894930b0b089a2459a88706d1407cf975b09"
       },
-      "version": "4.1.100.Final"
+      "version": "4.1.119.Final"
     },
     "io.netty:netty-tcnative-boringssl-static": {
       "shasums": {
@@ -621,45 +633,45 @@
     },
     "io.netty:netty-transport": {
       "shasums": {
-        "jar": "b1deeceedab3734cdb959c55f4be5ab4a667a8aed59121ff93763f49470f5470"
+        "jar": "c0921ea0e79c3b936daabe08a26e926642a4ad24951d6115fb43362c21dfeb10"
       },
-      "version": "4.1.100.Final"
+      "version": "4.1.119.Final"
     },
     "io.netty:netty-transport-classes-epoll": {
       "shasums": {
-        "jar": "9abc4b17b1212b33666eae4e8013d0bb78a9a2bcd0a9a621b9bd06a7e5fc0050"
+        "jar": "38edbceb4d0f3c1c87bc8db9f812d317342c9c8e95a4a052fb796a2f78ad098e"
       },
-      "version": "4.1.100.Final"
+      "version": "4.1.119.Final"
     },
     "io.netty:netty-transport-classes-kqueue": {
       "shasums": {
-        "jar": "70b40962b01ab594bed5cd311cd8768b224155c948118af60412ab23bc55ed50"
+        "jar": "af84f71a0bcde9e4ca2d8f1cb7bf6a6d4d249923a79ace14a86fea818a8f90f7"
       },
-      "version": "4.1.100.Final"
+      "version": "4.1.119.Final"
     },
     "io.netty:netty-transport-native-epoll": {
       "shasums": {
-        "linux-aarch_64": "81993265c3fb9098c0163943e71c14fbaa6ed17e345e7bf8205c90acfdf6cef5",
-        "linux-x86_64": "a108fb47babb7678b5b2abb1fc8b34510bd2f705faa450447860647f2de4ebaa"
+        "linux-aarch_64": "225aef9562189662e03c92bd5e5b3eb851a5499d87ef902b264cb924988b187d",
+        "linux-x86_64": "12557923a5069e736a9305c73a87b303ba9e89b32da3ec95dea6a856f3fd6397"
       },
-      "version": "4.1.100.Final"
+      "version": "4.1.119.Final"
     },
     "io.netty:netty-transport-native-kqueue": {
       "shasums": {
-        "osx-aarch_64": "0fdc323ad2efd85343713fbce6e0c2aec27e6852659984d03a5414018e01815f",
-        "osx-x86_64": "9ca2ba0121e35ea25228cf27ec553279966ca0a7f9c61e33ad1d9ea50477360d"
+        "osx-aarch_64": "52d80683a7644cef4294a719d285cd68979a824319081e240d149cb2e65cc751",
+        "osx-x86_64": "bc3977b0cd56a3df61c6117a8df41dfd3aefee862e3c51ca0d4a38c333ff4aa0"
       },
-      "version": "4.1.100.Final"
+      "version": "4.1.119.Final"
     },
     "io.netty:netty-transport-native-unix-common": {
       "shasums": {
-        "jar": "5d888230a04c4a3e647c64e933cefb64fd49056f969bfb734c8a3fcedf0bea8a",
-        "linux-aarch_64": "adb619d38e0e3411c10c68934a3b9313da31e29ecb6123eaf88fafb25688812e",
-        "linux-x86_64": "14c2deedac8b5092a5740a04196dda76ea9291501f7d45cdc8420ef4fd95fdf3",
-        "osx-aarch_64": "411b2cfdbb108c26a1c585d9e3057151033cd3e770a0f4a8384df9aa519e6810",
-        "osx-x86_64": "51624a67425e1fb5563ea51817171c9bc39c60be59e0ba78f19584aacc01b693"
+        "jar": "8ae5618f69326e6d17b77b0140f9d277118dbd23906ca07638bc01365728c650",
+        "linux-aarch_64": "57465b49ebf1663c20f10e87b72eb78ad61d91a7416e460bdc56a01de42f00d6",
+        "linux-x86_64": "b248e4dc0f004ffc884ebf30db435d1889a3a2fa40779445f4b010c4fb272603",
+        "osx-aarch_64": "ca21d9b9e26236c39d6db0ceae6cd6f65e12539143eb649cb1d95f6208485ff3",
+        "osx-x86_64": "0dc7a530458d1cae4963c3efc1bde9aa4727e2bba66dfa4ea0f3170725f26a5a"
       },
-      "version": "4.1.100.Final"
+      "version": "4.1.119.Final"
     },
     "io.opencensus:opencensus-api": {
       "shasums": {
@@ -2577,8 +2589,12 @@
       "io.netty.util.concurrent",
       "io.netty.util.internal",
       "io.netty.util.internal.logging",
+      "io.netty.util.internal.shaded.org.jctools.counters",
+      "io.netty.util.internal.shaded.org.jctools.maps",
       "io.netty.util.internal.shaded.org.jctools.queues",
       "io.netty.util.internal.shaded.org.jctools.queues.atomic",
+      "io.netty.util.internal.shaded.org.jctools.queues.atomic.unpadded",
+      "io.netty.util.internal.shaded.org.jctools.queues.unpadded",
       "io.netty.util.internal.shaded.org.jctools.util",
       "io.netty.util.internal.svm"
     ],
-- 
2.50.1

